zerver/tests/test_home.py

```diff
@@ -736,83 +746,100 @@ def test_show_billing(self) -> None:
+        # member, with CustomerPlan and realm plan_type STANDARD -> neither billing link or plans
         user.is_billing_admin = False
         user.save(update_fields=["is_billing_admin"])
-        self.assertNotInHomePage("Billing")
+        with self.settings(CORPORATE_ENABLED=True):
+            billing_info = get_billing_info(user)
+        self.assertFalse(billing_info.show_billing)
+        self.assertFalse(billing_info.show_plans)
 
-        # guest, with CustomerPlan -> no billing link
+        # guest, with CustomerPlan and realm plan_type SELF_HOSTED -> neither billing link or plans
         user.role = UserProfile.ROLE_GUEST
         user.save(update_fields=["role"])
-        self.assertNotInHomePage("Billing")
+        do_change_plan_type(user.realm, Realm.SELF_HOSTED, acting_user=None)
+        with self.settings(CORPORATE_ENABLED=True):
+            billing_info = get_billing_info(user)
+        self.assertFalse(billing_info.show_billing)
+        self.assertFalse(billing_info.show_plans)
 
-        # billing admin, but no CustomerPlan -> no billing link
+        # billing admin, but no CustomerPlan and realm plan_type SELF_HOSTED -> neither billing link or plans
         user.role = UserProfile.ROLE_MEMBER
         user.is_billing_admin = True
         user.save(update_fields=["role", "is_billing_admin"])
         CustomerPlan.objects.all().delete()
-        self.assertNotInHomePage("Billing")
+        with self.settings(CORPORATE_ENABLED=True):
+            billing_info = get_billing_info(user)
+        self.assertFalse(billing_info.show_billing)
+        self.assertFalse(billing_info.show_plans)
 
-        # billing admin, with sponsorship pending -> show billing link
+        # billing admin, with sponsorship pending and relam plan_type SELF_HOSTED -> show only billing link
         customer.sponsorship_pending = True
         customer.save(update_fields=["sponsorship_pending"])
-        self.assertInHomePage("Billing")
+        with self.settings(CORPORATE_ENABLED=True):
+            billing_info = get_billing_info(user)
+        self.assertTrue(billing_info.show_billing)
+        self.assertFalse(billing_info.show_plans)
 
-        # billing admin, no customer object -> make sure it doesn't crash
+        # billing admin, no customer object and relam plan_type SELF_HOSTED -> neither billing link or plans
         customer.delete()
-        result = self._get_home_page()
-        self.check_rendered_logged_in_app(result)
-
-    def test_show_plans(self) -> None:
-        realm = get_realm("zulip")
-
-        # Don't show plans to guest users
-        self.login("polonius")
-        do_change_plan_type(realm, Realm.LIMITED, acting_user=None)
-        self.assertNotInHomePage("Plans")
+        with self.settings(CORPORATE_ENABLED=True):
+            billing_info = get_billing_info(user)
+        self.assertFalse(billing_info.show_billing)
+        self.assertFalse(billing_info.show_plans)
```

Comment:
We can do a `get_billing_info(None)` test once we've added logic for that case in a prep commit.